src/environment.rs

```diff
@@ -76,7 +79,44 @@ impl Environment {
 		*self.env.lock().unwrap().env_ptr.get_mut()
 	}
 
-	fn new(name: String, log_level: LoggingLevel, execution_providers: Vec<ExecutionProvider>) -> OrtResult<Environment> {
+	fn create_custom_log_env(name: &str, log_level: LoggingLevel, _: HashMap<String, String>) -> (OrtStatusPtr, *mut sys::OrtEnv) {
+		let mut env_ptr: *mut sys::OrtEnv = std::ptr::null_mut();
+		let logging_function: sys::OrtLoggingFunction = Some(custom_logger);
+		// FIXME: What should go here?
+		let logger_param: *mut std::ffi::c_void = std::ptr::null_mut();
+		let cname = CString::new(name.clone()).unwrap();
+		let create_env_with_custom_logger = ortsys![CreateEnvWithCustomLogger];
+		let status = unsafe { create_env_with_custom_logger(logging_function, logger_param, log_level.into(), cname.as_ptr(), &mut env_ptr) };
+		(status, env_ptr)
+	}
+
+	fn create_global_thread_pool_env(name: &str, log_level: LoggingLevel, mut options: HashMap<String, String>) -> (OrtStatusPtr, *mut sys::OrtEnv) {
+		let mut env_ptr: *mut sys::OrtEnv = std::ptr::null_mut();
+		let mut thread_options: *mut sys::OrtThreadingOptions = std::ptr::null_mut();
+		let cname = CString::new(name.clone()).unwrap();
+		let create_thread_options = ortsys![CreateThreadingOptions];
+		let release_thread_options = ortsys![ReleaseThreadingOptions];
+		let create_env_with_global_thread_pool = ortsys![CreateEnvWithGlobalThreadPools];
```

Comment:
[`CreateEnvWithCustomLoggerAndGlobalThreadPools`](https://onnxruntime.ai/docs/api/c/struct_ort_api.html#a71336a3b27040097136a6fb40cf056c1)?

Reply:
oh sure

Reply:
ok fixed.. not sure why the pr still shows the old code